File: src/main/java/com/prasad/nithin/xmas/entity/UserGiftPreference.java
/**
 * 
 */
package com.prasad.nithin.xmas.entity;

import java.io.Serializable;
import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.ElementCollection;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToOne;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * @author 982168
 *
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class UserGiftPreference implements Serializable{

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	String id;
	
	@OneToOne(cascade = CascadeType.ALL)
	private User user;
	@ElementCollection
	private List<String> preference;
	private String deliveryAddress;
	private String postalCode;
}

File: src/main/ng-app/xmas-ui/src/app/datamodel/UserGiftPreference.ts
import { Teams } from "./Teams";
import { User } from "./User";

export class UserGiftPreference {
    constructor(
        private _id: String,
        private _user: User,
        private _preference: String[],
        private _deliveryAddress: String,
        private _postalCode: String
    ) { }

    get Id() {
        return this._id;
    }

    set Id(id: String) {
        this._id = id;
    }
    get User() {
        return this._user;
    }

    set User(user: User) {
        this._user = user;
    }
    get Preference() {
        return this._preference;
    }

    set Preference(preference: String[]) {
        this._preference = preference;
    }
    get DeliveryAddress() {
        return this._id;
    }

    set DeliveryAddress(deliveryAddress: String) {
        this._deliveryAddress = deliveryAddress;
    }
    get postalCode() {
        return this._postalCode;
    }

    set PostalCode(id: String) {
        this._postalCode = id;
    }

    getRequestBody() {
        let data = {
            "user": this._user.getRequestObject(),
            "preference": this._preference,
            "deliveryAddress": this._deliveryAddress,
            "postalCode": this._postalCode
        }
        
        return data;
    }
}